cli/tests/upstream.rs

```diff
@@ -13,18 +13,31 @@ async fn upstream_sync() -> TestResult {
 
     // Set up the test harness
     let test = Test::using_fixture("upstream.wasm")
-        .backend("origin", "http://127.0.0.1:9000/")
+        .backend("origin", "http://127.0.0.1:9000/", None)
         // The "origin" backend simply echos the request body
         .host(9000, |req| {
             let body = req.into_body();
             Response::new(body)
         })
         // The "prefix-*" backends return the request URL as the response body
-        .backend("prefix-hello", "http://127.0.0.1:9001/hello")
-        .backend("prefix-hello-slash", "http://127.0.0.1:9001/hello/")
+        .backend("prefix-hello", "http://127.0.0.1:9001/hello", None)
+        .backend("prefix-hello-slash", "http://127.0.0.1:9001/hello/", None)
         .host(9001, |req| {
             let body = req.uri().to_string().into_bytes();
             Response::new(body)
+        })
+        // The "override-host" backend checks the Host header
+        .backend(
+            "override-host",
+            "http://127.0.0.1:9002/",
+            Some("otherhost.com"),
+        )
+        .host(9002, |req| {
+            assert_eq!(
+                req.headers().get(header::HOST),
+                Some(&hyper::header::HeaderValue::from_static("otherhost.com"))
```

Comment:
Sorry, import nit: Could we remove the fully-specified identifier, and just use `HeaderValue` here? It's not ambiguous with other symbols here.

Reply:
```suggestion
                Some(&HeaderValue::from_static("otherhost.com"))
```

Reply:
Do you mean import HeaderValue directly? btw, http-semantics.rs uses the full-specified identifier.